plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -168,127 +169,129 @@ private void doVacuum(
         Instant threshold = Instant.now().minusMillis(retentionDuration.toMillis());
 
         DeltaLakeMetadata metadata = metadataFactory.create(session.getIdentity());
-        SchemaTableName tableName = new SchemaTableName(schema, table);
-        ConnectorTableHandle connectorTableHandle = metadata.getTableHandle(session, tableName);
-        checkProcedureArgument(connectorTableHandle != null, "Table '%s' does not exist", tableName);
-        DeltaLakeTableHandle handle = checkValidTableHandle(connectorTableHandle);
+        try (UncheckedCloseable ignore = () -> metadata.cleanupQuery(session)) {
```

Comment:
I counted 18 appearances of `implements Provider<Procedure>` in the code base.
Did you leave out intentionally the ones from:
- trino-base-jdbc
- trino-iceberg
- trino-main
?

Reply:
@findinpath good point. Not all procedures need update. Only those dealing with `ConnectorMetadata`.
Do you see any procedures that still require an update?

Reply:
Nothing else found to be changed.